cursor.go

```diff
@@ -521,3 +510,110 @@ func (c *Cursor) extendLocked(response *Response) {
 
 	putResponse(response)
 }
+
+// seekCursor takes care of loading more data if needed and applying pending skips
+//
+// bufferResponse determines whether the response will be parsed into the buffer
+func (c *Cursor) seekCursor(bufferResponse bool) error {
+	if c.lastErr != nil {
+		return c.lastErr
+	}
+
+	if len(c.responses) == 0 && c.closed {
```

Comment:
I had a couple of issues with this bit of code, what do you think about this change?

`if len(c.buffer) == 0 && len(c.responses) == 0 && c.closed {`